UnitsNet/BaseUnits.cs

```diff
@@ -0,0 +1,177 @@
+        public BaseUnits(LengthUnit length, MassUnit mass, DurationUnit time, ElectricCurrentUnit current,
+            TemperatureUnit temperature, AmountOfSubstanceUnit amount, LuminousIntensityUnit luminousIntensity)
+        {
+            Length = length;
+            Mass = mass;
+            Time = time;
+            Current = current;
+            Temperature = temperature;
+            Amount = amount;
+            LuminousIntensity = luminousIntensity;
+        }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            if(obj is null || !(obj is BaseUnits))
+                return false;
+
+            return Equals((BaseUnits)obj);
+        }
+
+        /// <summary>
+        /// Checks if all of the base units are equal to another instance's.
+        /// </summary>
+        /// <param name="other">The other instance to check if equal to.</param>
+        /// <returns>True if equal, otherwise false.</returns>
+#if WINDOWS_UWP
+        [Windows.Foundation.Metadata.DefaultOverload]
+#endif
+        public bool Equals(BaseUnits other)
+        {
+            if(other is null)
+                return false;
+
+            return Length == other.Length &&
+                Mass == other.Mass &&
+                Time == other.Time &&
+                Current == other.Current &&
+                Temperature == other.Temperature &&
+                Amount == other.Amount &&
+                LuminousIntensity == other.LuminousIntensity;
+        }
+
+        /// <inheritdoc />
+        public override int GetHashCode()
+        {
+            return new {Length, Mass, Time, Current, Temperature, Amount, LuminousIntensity}.GetHashCode();
+        }
+
+#if !WINDOWS_UWP
+
+        /// <summary>
+        /// Checks if this instance is equal to another.
+        /// </summary>
+        /// <param name="left">The left instance.</param>
+        /// <param name="right">The right instance.</param>
+        /// <returns>True if equal, otherwise false.</returns>
+        /// <seealso cref="Equals(BaseUnits)"/>
+        public static bool operator ==(BaseUnits left, BaseUnits right)
+        {
+            return left is null ? right is null : left.Equals(right);
+        }
+
+        /// <summary>
+        /// Checks if this instance is not equal to another.
+        /// </summary>
+        /// <param name="left">The left instance.</param>
+        /// <param name="right">The right instance.</param>
+        /// <returns>True if not equal, otherwise false.</returns>
+        /// <seealso cref="Equals(BaseUnits)"/>
+        public static bool operator !=(BaseUnits left, BaseUnits right)
+        {
+            return !(left == right);
+        }
+
+#endif
+
+        /// <inheritdoc />
+        public override string ToString()
+        {
+            var sb = new StringBuilder();
+
+            sb.AppendFormat("[Length]: {0}, ", UnitAbbreviationsCache.Default.GetDefaultAbbreviation(Length));
```

Comment:
Is abbreviation or dimensional number more useful? I don't know myself.

Reply:
These are the actual units, so the abbreviation is appropiate.